integration/inspektor-gadget/integration_test.go

```diff
@@ -585,10 +586,30 @@ func TestFiletop(t *testing.T) {
 	t.Parallel()
 
 	filetopCmd := &Command{
-		Name:           "StartFiletopGadget",
-		Cmd:            fmt.Sprintf("$KUBECTL_GADGET top file -n %s", ns),
-		ExpectedRegexp: fmt.Sprintf(`%s\s+test-pod\s+test-pod\s+\d+\s+\S*\s+0\s+\d+\s+0\s+\d+\s+R\s+date`, ns),
-		StartAndStop:   true,
+		Name:         "StartFiletopGadget",
+		Cmd:          fmt.Sprintf("$KUBECTL_GADGET top file -n %s -o json", ns),
+		StartAndStop: true,
+		ExpectedOutputFn: func(output string) error {
+			expectedEntry := &filetopTypes.Stats{
+				CommonData: BuildCommonData(ns),
+				Reads:      0,
+				ReadBytes:  0,
+				Filename:   "date.txt",
+				FileType:   byte('R'), // Regular file
+			}
+
+			normalize := func(e *filetopTypes.Stats) {
+				e.Node = ""
+				e.Writes = 0
+				e.WriteBytes = 0
+				e.Pid = 0
+				e.Tid = 0
+				e.MountNsID = 0
+				e.Comm = ""
```

Comment:
IMO, we could verify that the command is `sh` as far as we continue using [`BusyboxPodRepeatCommand`](https://github.com/inspektor-gadget/inspektor-gadget/blob/7167da66277a05ba3a617180d17e9cdb54407a64/integration/command.go#L507-L509) to start the pod and [`echo`](https://github.com/inspektor-gadget/inspektor-gadget/blob/7167da66277a05ba3a617180d17e9cdb54407a64/integration/inspektor-gadget/integration_test.go#L618) to write into the file.